gluten-ut/test/src/test/scala/org/apache/gluten/extension/GlutenExtensionRewriteRuleSuite.scala

```diff
@@ -62,4 +62,21 @@ class GlutenExtensionRewriteRuleSuite extends WholeStageTransformerSuite {
       }
     )
   }
+
+  test("GLUTEN-9279 - Not Pull out expression to avoid invalid reference binding") {
+    withTable("t") {
+      sql("CREATE TABLE t(f1 String, f2 String, f3 String, f4 String) USING CSV")
+      sql("INSERT INTO t values ('1', '2', '3', '4'), ('11' ,'22', '33', '4')")
+      runQueryAndCompare(
+        """
+          |SELECT SUM(f1) / COUNT(DISTINCT f2, f3) FROM t GROUP BY f4;
+          |""".stripMargin,
+        noFallBack = false
+      )(
+        df => {
+          checkGlutenOperatorMatch[ProjectExecTransformer](df)
```

Comment:
Also check the number of ProjectExecTransformer and HashAggregateExecBaseTransformer.